presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitSource.java

```diff
@@ -482,6 +497,11 @@ void fail(Throwable e)
 
                 if (internalSplit.isDone()) {
                     removedEstimatedSizeInBytes += internalSplit.getEstimatedSizeInBytes();
+
+                    // rewindable split sources keep their splits forever
+                    if (!useRewindableSplitSource && queues.decrementAndGetPartitionReferences(internalSplit) == 0) {
+                        estimatedSplitSizeInBytes.addAndGet(-internalSplit.getPartitionInfo().getEstimatedSizeInBytes());
```

Comment:
nit: simply increment the `removedEstimatedSizeInBytes`, same as for the internalSplit memory usage